boards/Makefile.include.cortexm_common

```diff
@@ -1,25 +1,59 @@
 # Target triple for the build. Use arm-none-eabi if you are unsure.
 export TARGET_TRIPLE ?= arm-none-eabi
 
-# Toolchain prefix, defaults to target triple followed by a dash, you will most likely not need to touch this.
+# Use TOOLCHAIN environment variable to select the toolchain to use.
+ifeq (,$(TOOLCHAIN))
+# Test if CC was set by the user or if by default from make
+ifneq (default,$(origin CC))
+ifeq (clang,$(notdir $(CC)))
+TOOLCHAIN = llvm
+else ifneq (,$(filter %gcc,$(CC)))
+TOOLCHAIN = gnu
+# Derive the prefix from the CC variable
+PREFIX = $(patsubst %gcc,%,$(CC))
+endif
+endif
+endif
+
+# TOOLCHAIN = clang is an alias for TOOLCHAIN = llvm
+ifeq (clang,$(TOOLCHAIN))
+TOOLCHAIN = llvm
+endif
+
+ifeq (,$(TOOLCHAIN))
+# Default: gnu
+TOOLCHAIN = gnu
+endif
+
+export TOOLCHAIN
+
+ifeq (,$(PREFIX))
+# default toolchain prefix, defaults to target triple followed by a dash, you
+# will most likely not need to touch this.
 export PREFIX ?= $(if $(TARGET_TRIPLE),$(TARGET_TRIPLE)-)
+endif
 
-# we build all cortex boards with the GNU toolchain
-include $(RIOTBOARD)/Makefile.include.gnu
+include $(RIOTBOARD)/Makefile.include.$(TOOLCHAIN)
 
 # define build specific options
-export CFLAGS_CPU   = -mcpu=$(MCPU) -mlittle-endian -mthumb -mno-thumb-interwork $(CFLAGS_FPU)
+export CFLAGS_CPU   = -mcpu=$(MCPU) -mlittle-endian -mthumb $(CFLAGS_FPU)
```

Comment:
If we move it, I think it will be easily overlooked that there are two places to modify if we need to change any flags. Also, it makes reading the files more tedious (switching back and forth between two/three files in the editor)